src/binding/transaction_log.cpp

```diff
@@ -0,0 +1,190 @@
+ * @param info - The callback info.
+ * @returns The new `NativeTransactionLog` object.
+ *
+ * @example
+ * ```ts
+ * const db = RocksDatabase.open('/tmp/testdb');
+ * db.open('/tmp/testdb');
+ * const txnLog = new NativeTransactionLog(db, 'test');
+ * ```
+ */
+napi_value TransactionLog::Constructor(napi_env env, napi_callback_info info) {
+	NAPI_CONSTRUCTOR_ARGV_WITH_DATA("TransactionLog", 2)
+
+	napi_ref exportsRef = reinterpret_cast<napi_ref>(data);
+	NAPI_GET_DB_HANDLE(argv[0], exportsRef, dbHandle, "Invalid argument, expected Database instance")
+
+	NAPI_GET_STRING(argv[1], name, "Transaction log store name is required")
+
+	std::shared_ptr<TransactionLogHandle>* txnLogHandle = new std::shared_ptr<TransactionLogHandle>(
+		std::make_shared<TransactionLogHandle>(*dbHandle, name)
+	);
+
+	DEBUG_LOG("TransactionLog::Constructor Creating NativeTransactionLog TransactionLogHandle=%p\n", txnLogHandle->get())
+
+	NAPI_STATUS_THROWS(::napi_wrap(
+		env,
+		jsThis,
+		reinterpret_cast<void*>(txnLogHandle),
+		[](napi_env env, void* data, void* hint) {
+			DEBUG_LOG("TransactionLog::Constructor NativeTransactionLog GC'd txnLogHandle=%p\n", data)
+			auto* txnLogHandle = static_cast<std::shared_ptr<TransactionLogHandle>*>(data);
+			if (txnLogHandle) {
+				(*txnLogHandle)->close();
+				txnLogHandle->reset();
+				delete txnLogHandle;
+			}
+		},
+		nullptr, // finalize_hint
+		nullptr  // result
+	));
+
+	return jsThis;
+}
+
+/**
+ * Adds an entry to the transaction log.
+ */
+napi_value TransactionLog::AddEntry(napi_env env, napi_callback_info info) {
+	NAPI_METHOD_ARGV(2)
+	UNWRAP_TRANSACTION_LOG_HANDLE("AddEntry")
+
+	// log entry
+	bool isBuffer;
+	NAPI_STATUS_THROWS(::napi_is_buffer(env, argv[0], &isBuffer));
+	bool isArrayBuffer;
+	NAPI_STATUS_THROWS(::napi_is_arraybuffer(env, argv[0], &isArrayBuffer));
+	if (!isBuffer && !isArrayBuffer) {
+		::napi_throw_error(env, nullptr, "Invalid log entry, expected a Buffer or Uint8Array");
+		return nullptr;
+	}
+
+	char* data = nullptr;
+	size_t size = 0;
+	NAPI_STATUS_THROWS(::napi_get_buffer_info(env, argv[0], reinterpret_cast<void**>(&data), &size));
+	if (data == nullptr) {
+		::napi_throw_error(env, nullptr, "Invalid log data, expected a Buffer or Uint8Array");
+		return nullptr;
+	}
+
+	// transaction (only required if TransactionHandle isn't bound to a transaction)
+	napi_valuetype type;
+	NAPI_STATUS_THROWS(::napi_typeof(env, argv[1], &type));
+	uint32_t transactionId = (*txnLogHandle)->transactionId;
+	if (type != napi_undefined) {
+		if (type == napi_number) {
+			NAPI_STATUS_THROWS(::napi_get_value_uint32(env, argv[1], &transactionId));
+		} else {
+			transactionId = 0;
+		}
+	}
+	if (transactionId == 0) {
+		::napi_throw_error(env, nullptr, "Invalid argument, expected a transaction id");
+		return nullptr;
+	}
+
+	// // create a reference to pin the buffer in memory (prevents GC)
+	napi_ref bufferRef = nullptr;
+	NAPI_STATUS_THROWS(::napi_create_reference(env, argv[0], 1, &bufferRef));
```

Comment:
I think we should copy the contents of the buffer into `data`, keep this as managed memory in C (delete when done) rather than maintaining a JS ref:
* Remove the complexity of managing, tracking, and deleting the JS reference
* Allow the JS buffer to be immediately GC'ed. For very large txns, helps protect against JS OOM, better scalability.
* Or allow for the safe reuse of shared buffer, that can be written over and over. Right now, I think this expects that the buffer won't be modified until the commit takes place, but this shouldn't be assumed, copying the data keeps it safe.